framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/deployment/RouterUtil.java

```diff
@@ -41,40 +34,22 @@
     public static void waitForRouter(URL url) {
         Instant endTime = Instant.now().plus(ROUTER_WAIT_TIME);
 
-        logger.info("Waiting for router to expose url: {}", url.toString());
+        String urlString = url.toString();
+        logger.info("Waiting for router to expose url: {}", urlString);
 
         while (Instant.now().isBefore(endTime)) {
             try {
-                HttpGet request = new HttpGet(url.toString());
-                HttpClient client = HttpClientBuilder.create().build();
-                HttpResponse response = client.execute(request);
-
-                if (response.getStatusLine().getStatusCode() != ROUTER_CODE) {
-                    return;
-                }
-
-                String responseContent = IOUtils.toString(response.getEntity().getContent(), "UTF-8");
-                if (!responseContent.contains(ROUTER_MESSAGE)) {
+                if (Https.getCode(urlString) != ROUTER_CODE) {
```

Comment:
I think it would be good what type of code we check. Change the name of is enough. e.g.:
```suggestion
                if (Https.getCode(urlString) != ROUTER_SERVICE_UNAVAILABLE_CODE) {
```

Reply:
Good point, updated.